sporkfish/searcher/minimax.py

```diff
@@ -130,6 +142,24 @@ def _update_killer_moves(self, move: chess.Move, depth: int) -> None:
             self._killer_moves[depth].pop()
             self._killer_moves[depth].insert(0, move)
 
+    def _update_history_table(self, move: chess.Move, depth: int) -> None:
+        """
+        Update the history table by incrementing the score of a move. This should
+        be called when a move causes an alpha-beta cutoff.
+
+        :param move: The move that caused an alpha-beta cutoff.
+        :type move: chess.Move
+        :param depth: The depth at which the move caused the cutoff
+        :type depth: int
+        """
+        increment = (self.max_depth - depth) * (self.max_depth - depth)
+        if move in self._history_table:
```

Comment:
Ah, I see - we might need 
if self._history_table and move in self._history_table:

Reply:
Oops yes indeed

Reply:
it was checked in line 79-85 when defining self._history_table:
`if (
    self._searcher_config.move_order_config.move_order_mode
    == MoveOrderMode.HISTORY
    or self._searcher_config.move_order_config.move_order_mode
    == MoveOrderMode.COMPOSITE
    ):
    self._history_table: Dict[chess.Move, int] = dict()`
we need to check this every time we do something with the history table?

Reply:
Yes - the problem is that that variable is not defined. Please see my reply on your mypy issue.